src/controllers/project.controller/updateProject/index.js

```diff
@@ -19,13 +21,81 @@ const updateProject = catchAsync(async (req, res) => {
     },
     changes: {},
   };
-  // if (req.body.caption) variables['changes']['caption'] = req.body.caption;
+
   if (req.body.description) variables.changes.description = req.body.description;
   if (req.body.title) variables.changes.title = req.body.title;
   if (req.body.link) variables.changes.link = req.body.link;
+  if (req.body.status !== undefined) variables.changes.status = req.body.status;
+  if (req.body.completed !== undefined) variables.changes.completed = req.body.completed;
 
   const response = await Hasura(updatePost_query, variables);
 
+  if (req.body.roles_required) {
+    const rolesUpdateVariables = {
+      projectId: id,
+      newRoles: req.body.roles_required.map((role) => ({
+        project_id: id,
+        role_name: role.role_name,
+      })),
+    };
+    await Hasura(updateRolesRequired, rolesUpdateVariables);
+  }
+
+  if (req.body.looking_for_mentors !== undefined || req.body.looking_for_members !== undefined) {
+    const projectFlagsUpdateVariables = {
+      projectId: id,
+      lookingForMentors: req.body.looking_for_mentors,
+      lookingForMembers: req.body.looking_for_members,
+    };
+    await Hasura(updateProjectFlags, projectFlagsUpdateVariables);
+  }
+
+  if (req.body.mentions) variables.changes.mentions = req.body.mentions;
+  if (req.body.project_tags) variables.changes.project_tags = req.body.project_tags;
+  if (req.body.documents) variables.changes.documents = req.body.documents;
+
+  if (req.body.looking_for_members || req.body.looking_for_mentors) {
+    const teamName = req.body.team_name ? req.body.team_name : `${req.body.title} team`;
+    const teamOnInovact = req.body.team_on_inovact;
+    const teamCreated = await createDefaultTeam(id, teamName, req.body.looking_for_mentors, req.body.looking_for_members, teamOnInovact);
+    variables.changes.team_id = teamCreated.team_id;
+  } else if (req.body.looking_for_members === false && req.body.looking_for_mentors === false) {
+    variables.changes.team_id = null;
```

Comment:
Inside this else block along with setting team_id null, we will have to delete that team from the teams table.